### PR TITLE
first commit in branch to test

### PR DESCRIPTION
**Pull request by ManuelPellegrini**
Setup of the user profile

### CODE REVIEW

**Comment by ayman-mezghani** [comment]
@samuelchassot @Gondolav We're still experiencing issues with the Jacoco coverage. It's been a week now. We followed the instruction on the related issue on the moodle forum: we made sure to update the paths (one for the coverage and one for the debugTree) in the build.gradle file, and we're running Jacoco with this command `./gradlew build connectedCheck jacocoTestReport`. Still the issue persists. When running the tests with coverage from Android Studio, the coverage for the concerned classes is above 90%.
Can you please help us.
Thanks in advance.

**Comment by samuelchassot** [comment]
> @samuelchassot @Gondolav We're still experiencing issues with the Jacoco coverage. It's been a week now. We followed the instruction on the related issue on the moodle forum: we made sure to update the paths (one for the coverage and one for the debugTree) in the build.gradle file, and we're running Jacoco with this command `./gradlew build connectedCheck jacocoTestReport`. Still the issue persists. When running the tests with coverage from Android Studio, the coverage for the concerned classes is above 90%.
> Can you please help us.
> Thanks in advance.

Hi,

I just ran the tests on my machine, using ```./gradlew build connectedCheck jacocoTestReport```
and in the report at *MusiConnect/app/build/reports/jacoco/jacocoTestReport/html/index.html* I found the same coverage as on CodeClimate. So the problem is that some tests are not taken into account for the report.
I am looking into it.
Does it work on other branches?

**Comment by ayman-mezghani** [comment]
> > @samuelchassot @Gondolav We're still experiencing issues with the Jacoco coverage. It's been a week now. We followed the instruction on the related issue on the moodle forum: we made sure to update the paths (one for the coverage and one for the debugTree) in the build.gradle file, and we're running Jacoco with this command `./gradlew build connectedCheck jacocoTestReport`. Still the issue persists. When running the tests with coverage from Android Studio, the coverage for the concerned classes is above 90%.
> > Can you please help us.
> > Thanks in advance.
> 
> Hi,
> 
> I just ran the tests on my machine, using `./gradlew build connectedCheck jacocoTestReport`
> and in the report at _MusiConnect/app/build/reports/jacoco/jacocoTestReport/html/index.html_ I found the same coverage as on CodeClimate. So the problem is that some tests are not taken into account for the report.
> I am looking into it.
> Does it work on other branches?

I believe that in the other branches, we're using only instrumented tests.

Thanks.

**Comment by samuelchassot** [comment]
> > > @samuelchassot @Gondolav We're still experiencing issues with the Jacoco coverage. It's been a week now. We followed the instruction on the related issue on the moodle forum: we made sure to update the paths (one for the coverage and one for the debugTree) in the build.gradle file, and we're running Jacoco with this command `./gradlew build connectedCheck jacocoTestReport`. Still the issue persists. When running the tests with coverage from Android Studio, the coverage for the concerned classes is above 90%.
> > > Can you please help us.
> > > Thanks in advance.
> > 
> > 
> > Hi,
> > I just ran the tests on my machine, using `./gradlew build connectedCheck jacocoTestReport`
> > and in the report at _MusiConnect/app/build/reports/jacoco/jacocoTestReport/html/index.html_ I found the same coverage as on CodeClimate. So the problem is that some tests are not taken into account for the report.
> > I am looking into it.
> > Does it work on other branches?
> 
> I believe that in the other branches, we're using only instrumented tests.
> 
> Thanks.

Okay so here, only instrumented tests are taken into account?

**Comment by samuelchassot** [comment]
> > > @samuelchassot @Gondolav We're still experiencing issues with the Jacoco coverage. It's been a week now. We followed the instruction on the related issue on the moodle forum: we made sure to update the paths (one for the coverage and one for the debugTree) in the build.gradle file, and we're running Jacoco with this command `./gradlew build connectedCheck jacocoTestReport`. Still the issue persists. When running the tests with coverage from Android Studio, the coverage for the concerned classes is above 90%.
> > > Can you please help us.
> > > Thanks in advance.
> > 
> > 
> > Hi,
> > I just ran the tests on my machine, using `./gradlew build connectedCheck jacocoTestReport`
> > and in the report at _MusiConnect/app/build/reports/jacoco/jacocoTestReport/html/index.html_ I found the same coverage as on CodeClimate. So the problem is that some tests are not taken into account for the report.
> > I am looking into it.
> > Does it work on other branches?
> 
> I believe that in the other branches, we're using only instrumented tests.
> 
> Thanks.

Okay, it took me quite some time but I finally figured it out. You imported the wrong thing for ```@Test``` as you can see in my review comment.
The *ExampleUnitTest* was actually taken into account but not yours and this is why. You imported a *Test* from *jupiter* and it is not the right one. Always be careful with automated imports, sometimes it imports something with the same name but from a fancy-exotic package and it can break the code and be a very painful error to find.
This corrected the problem locally but not on Travis.

The second issue was the path of the report in *.travis.yml*, it was the wrong one (as you should have noticed reading this post: https://moodle.epfl.ch/mod/forum/discuss.php?d=25904#p51415). I pointed it out in a comment too.

Now it should work. Let me know if it doesn't.

Cheers

File: app/src/test/java/ch/epfl/sdp/musiconnect/MusicianClassUnitTests.java [hunk 1]
```diff
@@ -0,0 +1,277 @@
+package ch.epfl.sdp.musiconnect;
+
+import org.junit.jupiter.api.Test;
```
Comment (samuelchassot) [inline app/src/test/java/ch/epfl/sdp/musiconnect/MusicianClassUnitTests.java]: This has to be ```import org.junit.Test``` (of course in every test file)

**Comment by ayman-mezghani** [comment]
> > > > @samuelchassot @Gondolav We're still experiencing issues with the Jacoco coverage. It's been a week now. We followed the instruction on the related issue on the moodle forum: we made sure to update the paths (one for the coverage and one for the debugTree) in the build.gradle file, and we're running Jacoco with this command `./gradlew build connectedCheck jacocoTestReport`. Still the issue persists. When running the tests with coverage from Android Studio, the coverage for the concerned classes is above 90%.
> > > > Can you please help us.
> > > > Thanks in advance.
> > > 
> > > 
> > > Hi,
> > > I just ran the tests on my machine, using `./gradlew build connectedCheck jacocoTestReport`
> > > and in the report at _MusiConnect/app/build/reports/jacoco/jacocoTestReport/html/index.html_ I found the same coverage as on CodeClimate. So the problem is that some tests are not taken into account for the report.
> > > I am looking into it.
> > > Does it work on other branches?
> > 
> > 
> > I believe that in the other branches, we're using only instrumented tests.
> > Thanks.
> 
> Okay, it took me quite some time but I finally figured it out. You imported the wrong thing for `@Test` as you can see in my review comment.
> The _ExampleUnitTest_ was actually taken into account but not yours and this is why. You imported a _Test_ from _jupiter_ and it is not the right one. Always be careful with automated imports, sometimes it imports something with the same name but from a fancy-exotic package and it can break the code and be a very painful error to find.
> This corrected the problem locally but not on Travis.
> 
> The second issue was the path of the report in _.travis.yml_, it was the wrong one (as you should have noticed reading this post: https://moodle.epfl.ch/mod/forum/discuss.php?d=25904#p51415). I pointed it out in a comment too.
> 
> Now it should work. Let me know if it doesn't.
> 
> Cheers

Thank you very much for the help. And yes we're aware that we didn't change the travis conf file yet, we were trying to make it work locally first.

File: app/src/test/java/ch/epfl/sdp/musiconnect/MusicianClassUnitTests.java (hunk 1 above)
Comment (ManuelPellegrini) [reply app/src/test/java/ch/epfl/sdp/musiconnect/MusicianClassUnitTests.java]: @samuelchassot 
At the beginning I tried to use `import org.junit.Test` as you suggest but then I wasn't able to use the `assertThrows()` method to test all the exception throws. Then I figured out that with the jupiter package I was able to test them, hence I changed the import.
If this is not allowed, then what should I do to test the exception throws?

Comment (samuelchassot) [reply app/src/test/java/ch/epfl/sdp/musiconnect/MusicianClassUnitTests.java]: Did you just try what I said here? Just replace all the ```import org.junit.jupiter.api.Test;``` by ```import org.junit.Test;```. That's it. No need to change your other imports.

**Pull request merged**